pyadjoint/drivers.py

```diff
@@ -69,3 +69,22 @@ def compute_hessian(J, m, m_dot, options=None, tape=None):
 
     r = [v.get_hessian(options=options) for v in m]
     return m.delist(r)
+
+
+def solve_adjoint(J, tape=None, adj_value=1.0):
+    """
+    Solve the adjoint problem for a functional J.
+
+    Adjoint actions are stored on each tape block as the `adj_value` attribute. In the case of
```

Comment:
Sorry, just to clarify: `adj_value`: is something that lives on a `BlockVariable` which represents a state of a Function/Constant/AdjFloat etc. not on blocks. The `adj_value` attribute on a block_variable accumulates the sensitivity of the functional wrt to that inbetween state when traversing backwards as you do here. the `adj_sol` attribute is something that lives on a solve block indeed. Maybe you want to say something like:

This traverses the entire tape backwards, unlike compute_gradient() which only works out those parts of the adjoint necessary to compute the sensitivity wrt the specified control. As a result sensitivities wrt to all inbetween states are accumulated in the adj_value attribute of the associated block-variables. The adjoint solution of each solution step is stored in the adj_sol attribute of the associated solve block.

Reply:
Thanks Stephan, have modified as suggested